config.c

```diff
@@ -26,6 +35,66 @@ void finish_config(struct mako_config *config) {
 	free(config->format);
 }
 
+/* Parse between 1 and 4 integers, comma separated, from the provided string.
+ * Depending on the number of integers provided, the four fields of the `out`
+ * struct will be initialized following the same rules as the CSS "margin"
+ * property. Returns 0 on success, -1 on parsing failure. */
+int parse_directional(const char *directional_string,
+		struct directional *out) {
+	int error = 0;
+	char *components = strdup(directional_string);
+
+	char *token, *saveptr, *endptr = NULL;
+	int32_t values[] = {0, 0, 0, 0};
+	int count = 0;
```

Comment:
Since this is a counter, it would make sense to make this a `size_t`. Also please define variables when there are used (here in the `for` loop).

Reply:
I can't move count into the for loop as it's also used down below. I did move `number` and `endptr` into the loop though.